addons/spreadsheet/static/src/global_filters/components/date_filter_dropdown/date_filter_dropdown.js

```diff
@@ -183,34 +184,37 @@ export class DateFilterDropdown extends Component {
     }
 
     update(value) {
+        let newFilterValue;
         switch (value.type) {
             case "relative":
-                this.props.update({ type: "relative", period: value.id });
+                newFilterValue = { type: "relative", period: value.id };
                 break;
             case "month":
-                this.props.update(this.selectedValues.month);
+                newFilterValue = this.selectedValues.month;
                 break;
             case "quarter":
-                this.props.update(this.selectedValues.quarter);
+                newFilterValue = this.selectedValues.quarter;
                 break;
             case "year":
-                this.props.update(this.selectedValues.year);
+                newFilterValue = this.selectedValues.year;
                 break;
             case "range": {
                 const { from, to } = this.selectedValues.range;
-                if (from && to) {
-                    // Ensure 'to' is after 'from'
-                    if (DateTime.fromISO(from) > DateTime.fromISO(to)) {
-                        this.selectedValues.range.to = from;
-                        this.selectedValues.range.from = to;
-                    }
+                if (from && to && DateTime.fromISO(from) > DateTime.fromISO(to)) {
+                    this.selectedValues.range.to = from;
+                    this.selectedValues.range.from = to;
                 }
-                this.props.update(this.selectedValues.range);
+                newFilterValue = this.selectedValues.range;
                 break;
             }
             default:
-                this.props.update(undefined);
+                newFilterValue = undefined;
         }
+
+        if (deepEqual(newFilterValue, this.props.value)) {
+            return;
```

Comment:
Shouldn't it be part of the `case "SET_GLOBAL_FILTER_VALUE"` in `allowDispatch` ?